### PR TITLE
Update Django from 3.2 to 4.0

### PR DESCRIPTION
**Pull request by vrigal**
Django update to 4.0.6 :
* `pytest-freezegun` dependency version has been fixed to https://github.com/ktosiek/pytest-freezegun/pull/38 (removes a distutils deprecation warning making tests fail).
* Remove USE_L10N django setting (defaults to True in 4.0).
* Set USE__TZ setting to False (defaults to True instead of false in 4.0, Treeherder only uses UTC times).
* Add `CSRF_TRUSTED_ORIGINS` setting, configurable with an environment variable. ⚠️ You'll have to define `CSRF_TRUSTED_ORIGINS` for deployments with origin URLs (e.g. frontend address) as defined in the [Django official documentation](https://docs.djangoproject.com/en/4.0/ref/settings/#std-setting-CSRF_TRUSTED_ORIGINS).

### CODE REVIEW

**Comment by vrigal** [comment]
Tests are not ran in the CI because of a deprecation warning from dockerflow dependency (ref. https://github.com/mozilla-services/python-dockerflow/issues/66).

Otherwise, there are 1 error and 2 fails. It should be possible to add a tearDown method as the failing tests run fine when executes separately.
```
E               Traceback (most recent call last):
E                 File "/usr/local/lib/python3.9/site-packages/celery/utils/collections.py", line 250, in __getitem__
E                   return mapping[_key]
E               ResourceWarning: unclosed <socket.socket fd=13, family=AddressFamily.AF_INET, type=SocketKind.SOCK_STREAM, proto=6, laddr=('172.26.0.5', 41964), raddr=('172.26.0.3', 5672)>
```

**Comment by jmaher** [comment]
@vrigal are you aware of anything we can do to fix the deprecated tools/packages/scripts in dockerflow?

**Comment by Archaeopteryx** [comment]
The python-dockerflow fix landed: https://github.com/mozilla-services/python-dockerflow/commit/0f640cc031aecf5d61432fc141719e5603d66acf

**Comment by jmaher** [comment]
look in dev.in and consider upgrading `django-debug-toolbar` and maybe others.

**Comment by vrigal** [comment]
I could update dev dependencies, but there is the same deprecation warning (`RemovedInDjango41Warning`) with the latest version of django-extensions. We could eventually separate development and tests related dependencies.

A more concerning issue another deprecation warning raised by `pytest-freezegun`, which is mentioned here :
* https://github.com/mozilla/treeherder/pull/7445#issue-1235147252]
* https://github.com/spulec/freezegun/issues/435

We could eventually run pytest with `-W ignore::DeprecationWarning`, or remove this dependency and skip a certain amount of tests

**Comment by vrigal** [comment]
In fact we can probably wait https://github.com/ktosiek/pytest-freezegun/pull/38 to be merged

File: treeherder/config/settings.py [hunk 1]
```diff
@@ -178,7 +178,9 @@
 # Internationalization
 TIME_ZONE = "UTC"
 USE_I18N = False
-USE_L10N = True
+
+# Timezones are not supported in Treeherder yet
+USE__TZ = False
```
Comment (La0) [inline treeherder/config/settings.py:183]: I guess it should be `USE_TZ`, single underscore

**Review by La0 — CHANGES_REQUESTED** [review]

**Review by jmaher — COMMENTED** [review]
the changes in general look good- curious why the line numbers on the test data had to change?

**Comment by vrigal** [comment]
> the changes in general look good- curious why the line numbers on the test data had to change?

Line numbers change in tests is related to the update of the responses package (works with `0.14.0` and fails with `0.15.0`).
It should not alter treeherder in my opinion, as it is a tests-related package.

**Comment by jmaher** [comment]
asking @La0 for review as there are pending requested/resolved changes.

**Review by jmaher — APPROVED** [review]

**Pull request merged**